File: COMPOSICION & AGREGACION/EJ11/Computadora.py
class Procesador:
    def __init__(self, marca, modelo, nucleos, velocidad):
        self.marca = marca
        self.modelo = modelo
        self.nucleos = nucleos
        self.velocidad = velocidad

class RAM:
    def __init__(self, marca, capacidad, frecuencia):
        self.marca = marca
        self.capacidad = capacidad
        self.frecuencia = frecuencia

class DiscoDuro:
    def __init__(self, marca, capacidad, tipo):
        self.marca = marca
        self.capacidad = capacidad
        self.tipo = tipo

class Computadora:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.procesador = Procesador("Intel", "i7-12700K", 12, "3.6GHz")
        self.ram = RAM("Corsair", 32, "3200MHz")
        self.disco_duro = DiscoDuro("Samsung", 1000, "SSD NVMe")
    
    def descripcion(self):
        print(f"Computadora: {self.marca} {self.modelo}")
        print(f"Procesador: {self.procesador.marca} {self.procesador.modelo} - {self.procesador.nucleos} núcleos - {self.procesador.velocidad}")
        print(f"RAM: {self.ram.marca} - {self.ram.capacidad}GB - {self.ram.frecuencia}")
        print(f"Disco Duro: {self.disco_duro.marca} - {self.disco_duro.capacidad}GB - {self.disco_duro.tipo}")

mi_pc = Computadora("Dell", "XPS 15")
mi_pc.descripcion()

File: COMPOSICION & AGREGACION/EJ11/Computadora.java
class Procesador {
    String marca;
    String modelo;
    int nucleos;
    String velocidad;
    
    Procesador(String marca, String modelo, int nucleos, String velocidad) {
        this.marca = marca;
        this.modelo = modelo;
        this.nucleos = nucleos;
        this.velocidad = velocidad;
    }
}

class RAM {
    String marca;
    int capacidad;
    String frecuencia;
    
    RAM(String marca, int capacidad, String frecuencia) {
        this.marca = marca;
        this.capacidad = capacidad;
        this.frecuencia = frecuencia;
    }
}

class DiscoDuro {
    String marca;
    int capacidad;
    String tipo;
    
    DiscoDuro(String marca, int capacidad, String tipo) {
        this.marca = marca;
        this.capacidad = capacidad;
        this.tipo = tipo;
    }
}

class Computadora {
    String marca;
    String modelo;
    Procesador procesador;
    RAM ram;
    DiscoDuro discoDuro;
    
    Computadora(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.procesador = new Procesador("Intel", "i7-12700K", 12, "3.6GHz");
        this.ram = new RAM("Corsair", 32, "3200MHz");
        this.discoDuro = new DiscoDuro("Samsung", 1000, "SSD NVMe");
    }
    
    void descripcion() {
        System.out.println("Computadora: " + this.marca + " " + this.modelo);
        System.out.println("Procesador: " + this.procesador.marca + " " + this.procesador.modelo + 
                          " - " + this.procesador.nucleos + " núcleos - " + this.procesador.velocidad);
        System.out.println("RAM: " + this.ram.marca + " - " + this.ram.capacidad + "GB - " + this.ram.frecuencia);
        System.out.println("Disco Duro: " + this.discoDuro.marca + " - " + this.discoDuro.capacidad + 
                          "GB - " + this.discoDuro.tipo);
    }
}

public class Main {
    public static void main(String[] args) {
        Computadora miPc = new Computadora("Dell", "XPS 15");
        miPc.descripcion();
    }
}